agatha/construct/semrep_util.py

```diff
@@ -302,49 +302,77 @@ def _get_flags(self, input_path:Path, output_path:Path)->List[str]:
     if self.word_sense_disambiguation:
       res.append("--word_sense_disambiguation")
     res.append("--xml_output_format")
-    res.append(str(input_path))
-    res.append(str(output_path))
+    if input_path is not None:
+      res.append(str(input_path))
+    if output_path is not None:
+      res.append(str(output_path))
     return res
 
-  def _run_gracefully(self, input_path:Path, output_path:Path)->None:
+  def _read_lines(self, in_file:Path)->List[str]:
     """
-    Runs semrep safely. If a sentence causes a semrep error, 
-    logs the error and re-tries execution with next input.
-    Overall, encapsulate run functionality and use @320 as a
-    wrapper.
+    Reads in lines of a file in memory
+
     Args:
-      input_path: Path to the Semrep input file
+     in_file: Path to input file
+    Returns:
+      lines in the files
     """
-    if not self.metamap_server.running():
-      self.metamap_server.start()
+    with open(in_file, 'r') as f:
+      lines = []
+      for line in f:
+        lines.append(line.strip())
+    return lines
+
 
-    input_path = Path(input_path)
-    assert input_path.is_file(), f"Failed to find {input_path}"
-    assert not output_path.exists(), f"Refusing to overwrite {output_path}"
-    cmd = self._get_flags(input_path, output_path)
-    env = self._get_env()
-    print("Running:", " ".join(cmd))
 
+  def _run_gracefully(self,
+      in_lines:List[str],
+      out_lines:List[lxml.etree._Element]
+      )->None:
+    """
+    Runs individual lines of the text through semrep and
+    produces the output xml. Raises a `ChildProcessError`
+    that is caught in the actual `run()` function.
+
+    Args:
+      lines: List[str]
+
+    Returns:
+      List of XML elements if no exception is raised
+    """
+    cmd = self._get_flags(None, None)
+    env = self._get_env()
     semrep_proc = subprocess.Popen(
         cmd,
+        env=env,
         stdin=subprocess.PIPE,
-        stdout=subprocess.PIPE)
-    _, error = semrep_proc.communicate()
+        stdout=subprocess.PIPE
+        )
+
+    out_lines, error = semrep_proc.communicate(
```

Comment:
Right now, this will overwrite the `out_lines` variable.  You won't get your expected behavior.
This function should return something like the `lxml` element.

Reply:
My idea was that if I passed in the list of `str` I should get a list of `xml` outputs. Does `semrep` work differently? Or maybe I could allocate a local variable and then return?

Reply:
I think the issue is that python doesn't work that way. The list that you pass in as input is not modified in this line.